sdl-deltalake/src/main/scala/io/smartdatalake/workflow/dataobject/DeltaLakeTableDataObject.scala

```diff
@@ -199,17 +200,65 @@ case class DeltaLakeTableDataObject(override val id: DataObjectId,
     }
   }
 
+  private def activateCdc()(implicit context: ActionPipelineContext): Unit = {
+    implicit val session: SparkSession = context.sparkSession
+    if(!propertyExists(EnableCdcFeedProperty) && isTableExisting) HiveUtil.alterTableProperties(table, Map(EnableCdcFeedProperty -> "true"))
+  }
+
+  private def propertyExists(name: String)(implicit session: SparkSession): Boolean = {
+    val details = DeltaTable.forName(session, table.fullName).detail()
+    val properties = details.select("properties").head.getMap[String, String](0)
+
+    properties.contains(name)
+  }
+
+  private def propertyExistsWithValue(name: String, value: String) (implicit session: SparkSession): Boolean = {
+    val details = DeltaTable.forName(session, table.fullName).detail()
+    val properties = details.select("properties").head.getMap[String, String](0)
+
+    properties.exists(_ == name -> value)
+  }
+
+  @transient private val EnableCdcFeedProperty = "delta.enableChangeDataFeed"
+
   override def getSparkDataFrame(partitionValues: Seq[PartitionValues] = Seq())(implicit context: ActionPipelineContext): DataFrame = {
-    val df = context.sparkSession.table(table.fullName)
+
+    implicit val session: SparkSession = context.sparkSession
+
+    val cdcActivated = propertyExistsWithValue(EnableCdcFeedProperty, "true")
+
+    val df = if(cdcActivated && incrementalOutputExpr.isDefined) {
+
+      require(table.primaryKey.isDefined, s"PrimaryKey for table [${table.fullName}] needs to be defined when using DataObjectStateIncrementalMode")
+
+      val windowSpec = Window.partitionBy(table.primaryKey.get.mkString(",")).orderBy(col("_commit_timestamp").desc)
+
+      context.sparkSession.read.format("delta")
+        .option("readChangeFeed", "true")
+        .option("startingVersion", incrementalOutputExpr.get)
+        .table(table.fullName)
+        .where(expr("_change_type IN ('insert','update_postimage')"))
+        .withColumn("rank", rank().over(windowSpec))
```

Comment:
What if the DataFrame already has a "rank" column?
I would prefix this with "_" as we do with most helper columns, e.g. "_rank"

Reply:
Fixed for both DeltaLakeTableDataObject and IcebergTableDataObject in 2aa23d7e0ba1977d76e85c551d068e5552978d7e